include/range/v3/utility/iterator.hpp

```diff
@@ -182,14 +189,32 @@ namespace ranges
                         --it;
                 return n;
             }
-            template<typename I, typename D, typename S, typename Concept>
+            template<typename I, typename D, typename S>
+            RANGES_CXX14_CONSTEXPR
+            D advance_fn::bounded_(I &it, D n, S bound, concepts::SizedIteratorRange*,
+                concepts::InputIterator*)
+            {
+                RANGES_ASSERT(0 <= n);
+                D d = bound - it;
+                RANGES_ASSERT(0 <= d);
+                if (d <= n)
+                {
+                    ranges::advance(it, std::move(bound));
+                    return n - d;
+                }
+                ranges::advance(it, n);
+                return 0;
+            }
+            template<typename I, typename D>
```

Comment:
This one and the next differed substantially in the first version of the patch, IIRC. The only real difference now is the type of `bound` and the `0 <= n` precondition on this overload. They could easily be merged into a single `I, D, S, SizedIteratorRange, Concept` overload with that precondition changed to `Same<I, S>() || 0 <= n`. That would also give me a chance to fix the space after `if` and the annoying fact that the first overload uses `d <= n` vs. the seconds `n >= d`.